src/Hudl.Mjolnir/Config/MjolnirConfiguration.cs

```diff
@@ -161,13 +162,14 @@ public void Dispose()
                 _onDispose();
             }
         }
-
-        private readonly List<IObserver<MjolnirConfiguration>> _observers = new List<IObserver<MjolnirConfiguration>>();
-
+        
+        private readonly ConcurrentDictionary<IObserver<MjolnirConfiguration>, IObserver<MjolnirConfiguration>>
+            _observers = new ConcurrentDictionary<IObserver<MjolnirConfiguration>, IObserver<MjolnirConfiguration>>();
+        
```

Comment:
Is there a better option than using the observer as both the key and the value? Not that I think this is wrong, just curious on alternatives.

Reply:
Joining the party a bit late but it seems like a `ConcurrentBag` would suffice here, seeing as you just add/remove.